File: backend/user-service-interface/src/main/java/app/user/api/user/RegisterUserRequest.java
package app.user.api.user;

import org.springframework.lang.NonNull;

/**
 * @author steve
 */
public class RegisterUserRequest {
    public String name;

    public Integer age;

    @NonNull
    public String email;

    @NonNull
    public String password;

    @NonNull
    public String requestedBy;
}

File: frontend/iblog-site-frontend/src/api/user/RegisterUserRequest.ts
export class RegisterUserRequest {
    public name: string | null;
    public age: number | null;
    public email: string;
    public password: string;

    constructor(name: string | null, age: number | null, email: string, password: string) {
        this.name = name;
        this.age = age;
        this.email = email;
        this.password = password;
    }
}
